extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpSecurityRecorder.java

```diff
@@ -47,6 +48,11 @@ public void run() {
                                         event.response().end();
                                     }
                                 });
+                            } else if (throwable instanceof AuthenticationRedirectException) {
```

Comment:
I don't think this is correct. As I understand it this is basically a challenge, so it should be send as part of sendChallenge()? 

Otherwise you loose the ability to have unprotected resources, as this will attempt to log in on every request even if the page is supposed to be public.

Reply:
@stuartwdouglas This exception may only be thrown on the return leg of the OIDC code flow, not when the user is initially trying to access a resource. (Ex, user accesses a protected resource, user is redirected, with some of the original path saved in the cookie, user is redirected back - and now if the original path needs to be restored the user will be redirect to it)
FYI, `CodeFlowTest` test accessing an `UnprotectedResource` which you added passes with this PR.

Reply:
ah, ok. In that case I think it is ok.